src/circ_deque.jl

```diff
@@ -80,6 +80,11 @@ end
     ret
 end
 
+# Iteration via getindex
+Base.start(d::CircularDeque) = 1
+Base.next(d::CircularDeque, i) = (d[i], i+1)
+Base.done(d::CircularDeque, i) = i == length(d) + 1
```

Comment:
I've pushed an "enhanced" version that uses a new `unsafe_getindex` to avoid bounds checking via getindex.